src/xunit.performance.api/XunitBenchmark.cs

```diff
@@ -14,23 +14,25 @@ static XunitBenchmark()
             // TODO: Read this from file.
             //   1. This reads as: flags read from config file based on running platform!
             //   2. Format could be something like this: `{ "Name" : "BranchMispredictions", "interval" : "100000" }`
-            RequiredProviders = new ProviderInfo[]
+            PerformanceMonitorCounters = new List<IPerformanceMonitorCounter>
+            {
+                new BranchMispredictionsPerformanceMonitorCounter(),
+                new CacheMissesPerformanceMonitorCounter(),
+            };
+
+            PerformanceMetricInfos = new List<BasePerformanceMetric>
```

Comment:
I'm guessing we need one `BasePerformanceMetric` for each `IPerformanceMonitorCounter` above, right? Could we somehow loop through the list and automatically create one, to avoid having two mirrored lists?